.travis.yml

```diff
@@ -1,27 +1,51 @@
 language: elixir
 elixir:
-  - 1.6.1
+  - 1.8.1
 otp_release:
-  - 20.2
+  - 21.3.2
 dist: trusty
 sudo: false
 services: true
 
-cache:
-  directories:
-    - _build
-    - deps
+jobs:
+  include:
+    - name: Linters
+      env: CACHE_NAME=linters
+      cache:
+        directories:
+          - _build
+          - deps
+      script:
+        - mix local.hex --force
+        - mix deps.unlock --unused && git diff --exit-code
+        - mix format --check-formatted
+        - mix deps.get --only test
+        - mix credo strict
 
-before_install: true
+    - name: Unit Tests
+      env: CACHE_NAME=unit_tests
+      cache:
+        directories:
+          - _build
+          - dep
+      before_script:
+        - mix local.hex --force
+        - mix deps.get --only test
+        - mix deps.clean --unused
+        - MIX_ENV=test mix compile --warnings-as-error
+      script:
+        - mix test
+        - mix coveralls.travis
 
-before_script:
-  - mix local.hex --force
-  - mix deps.get --only test
-  - mix deps.clean --unused
-  - MIX_ENV=test mix compile --warnings-as-errors
-  - MIX_ENV=test travis_wait mix dialyzer --plt
-
-script:
- - mix test
- - mix coveralls.travis
- - MIX_ENV=test mix dialyzer --halt-exit-status
\ No newline at end of file
+    - name: Dialyzer
+      env: CACHE_NAME=dialyzer
+      cache:
+        directories:
+          - _build
+          - deps
+      before_script:
+        - mix deps.clean --unused
+        - mix compile --warnings-as-errors
+        - travis_wait 30 mix dialyzer --plt
+      script:
+        - mix dialyzer --format short --halt-exit-status
```

Comment:
What would you think about cherry-picking these build updates into something that could be merged immediately? Or would dialyzer and credo not pass on the current codebase? I was just looking over the commit and thought bumping the elixir version didn't fit with "Fixes issues decoding Multipass payloads". However, if this is too much of a pain, I'm fine if you leave it in.

Reply:
I looked into the version specified within `mix.exs` and it's 1.6 so I will revert the travis version back to that. I believe this was a copy-pasta error from another TF repo.